src/comm/aca_grpc.cpp

```diff
@@ -41,9 +41,13 @@ HostRequestReply GoalStateProvisionerImpl::RequestGoalStates(HostRequest *reques
 {
   grpc::ClientContext ctx;
   alcor::schema::HostRequestReply reply;
-  
-  if (chan_->GetState(false) != grpc_connectivity_state::GRPC_CHANNEL_READY) {
-    ACA_LOG_INFO("%s, it is: [%d]\n", "Channel state is not READY", chan_->GetState(false));
+
+  // check current grpc channel state, try to connect if needed
+  grpc_connectivity_state current_state = chan_->GetState(true);
+  if (current_state == grpc_connectivity_state::GRPC_CHANNEL_SHUTDOWN ||
+      current_state == grpc_connectivity_state::GRPC_CHANNEL_TRANSIENT_FAILURE) {
+    ACA_LOG_INFO("%s, it is: [%d]\n",
```

Comment:
In addition to print out the log, does our code do retry or rely on gRPC to retry?

Reply:
Fixed